dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -2092,6 +2098,37 @@ public void deleteWorkflow(@ApiParam(hidden = true) @Parameter(hidden = true, na
         }
     }
 
+    @GET
+    @Path("/github/infer/{owner}/{repo}/{ref}")
+    @Timed
+    @UnitOfWork
+    @Operation(description = "Infer the entries in the file tree of a GitHub repository reference.", security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME))
+    @Deprecated
+    public String inferEntries(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user") @Auth User user,
```

Comment:
Do we  need  to think about permissions?
Does  this need a `checkCanRead`?

Reply:
This method doesn't access any entries or other possibly-protected Dockstore entities, it uses the user's GitHub token to submit the GitHub API requests with that user's privileges and rate limit.  It then infers the `.dockstore.yml` from the contents of the specified repo/ref.  If the GitHub user doesn't have access to that repo, it'll throw.  No `checkCanRead` necessary.